lib/decorators/on-event.decorator.ts

```diff
@@ -21,6 +22,22 @@ export interface OnEventMetadata {
  */
 export type OnEventType = string | symbol | Array<string | symbol>;
 
+/**
+ * Wraps the method in try/catch blocks.
+ * @param descriptor
+ */
+const wrapDescriptorInTryCatchBlocks = (descriptor: any) => {
+  const originMethod = descriptor.value;
```

Comment:
overriding the descriptor will consequently make other metadata (attached through `@SetMetadata`) lost for this method

Reply:
I expect to be able to maintain through line 38(`Object.setPrototypeOf(descriptor.value, originMethod);`).

Reply:
Wouldn't it be easier to just add try-catch here https://github.com/nestjs/event-emitter/blob/master/lib/event-subscribers.loader.ts#L95 and here https://github.com/nestjs/event-emitter/blob/c25acc940103738f445decf1b14ee23347845ba5/lib/event-subscribers.loader.ts#L138

Reply:
Thanks for the suggestion.